setup.sh

```diff
@@ -0,0 +1,283 @@
+            read -p "Press enter to continue"
+
+            export PATH=/usr/bin:$PATH
+            sudo -u postgres createuser --superuser $USER
+            sudo -u postgres createdb $USER
+        fi
+
+        psql -c "CREATE ROLE itig WITH CREATEDB PASSWORD 'd872\$dh' LOGIN;"
+        if [ $? -ne 0 ]; then
+            error "Could not install postgres. Please ensure psql is on the path, and then rerun this script."
+            exit 1
+        fi
+
+        verbose "installed postgres"
+}
+
+# 
+# Install native tools
+# 
+install_native_tools () {
+    msg "Installing native tools"
+    if isMac; then
+        brew install imagemagick libmagic ghostscript
+        msg "Trying to install pygments with easy_install, please enter your password"
+        sudo easy_install Pygments
+    else
+        sudo apt-get install -y ghostscript \
+                       imagemagick \
+                       libmagickwand-dev \
+                       libmagic-dev \
+                       python-pygments
+    fi
+    if [ $? -ne 0 ]; then
+        error "Could not install native tools, please review the terminal window for details."
+        error "Packages may have already been installed, attempting to continue with setup."
+    else verbose "installed native tools"
+    fi
+}
+
+# 
+# Install doubtfire gem dependencies
+# 
+install_dfire_dependencies () {
+    msg "Installing doubtfire dependencies"
+    gem install bundler
+    bundler install --without production replica
+    rbenv rehash
+    source ~/.bashrc
+
+    msg "Populating database"
+    rake db:create
+    rake db:populate
+
+    if [ $? -ne 0 ]; then
+        error "Could not populate database."
+        exit 1
+    fi
+
+    verbose "populated database"
+}
+
+# 
+# Install doubtfire gem dependencies
+# 
+install_dstil_overcommit () {
+    curl -s https://raw.githubusercontent.com/dstil/dotfiles/master/bootstrap | bash
+
+    gem install overcommit
+    rbenv rehash
+    overcommit --install
+    dstil --sign
+
+    verbose "installed dstil hooks."
+}
+
+install_latex () {
+    if isMac; then
+        read -r -p "LaTeX is required for PDF generation, it could take up to several hours to install, would you like to install LaTeX now? [y/N] " response
+        case $response in
+            [yY][eE][sS]|[yY]) 
+                brew cask install mactex
+                ;;
+            *)
+                msg "You will not be able to generate PDF's without LaTeX"
+                ;;
+        esac
+    else 
+        sudo apt-get install -y texlive-full
+    fi
+    verbose "Installed LaTeX"
+}
+
+if isMac; then
+    install_homebrew
+fi
+
+install_rbenv
+install_postgres
+install_native_tools
+install_dfire_dependencies
+install_dstil_overcommit
+install_latex
+exec $SHELL
+
+msg "You can now launch the server with rails s"
+
+verbose "Doubtfire developed successfuly installed!"
```

Comment:
This could probably be verbose. But we should only say it's successful if it was successful? E.g., if failure, then it will exit one of the functions, come out to here, but then we will still echo this?